bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,536 @@
+                                        properties.put("Supported vane up/down modes", swingModes.toString());
+                                        updateProperties(properties);
+                                        addChannel(channelId, itemType, swingModes);
+                                        break;
+                                    case 6:
+                                        channelId = CHANNEL_TYPE_VANESUD;
+                                        properties.put("Supported vane up/down modes", swingModes.toString());
+                                        updateProperties(properties);
+                                        addChannel(channelId, itemType, swingModes);
+                                        break;
+                                }
+                                break;
+                            case 9:
+                                channelId = CHANNEL_TYPE_TARGETTEMP;
+                                itemType = "Number:Temperature";
+                                addChannel(channelId, itemType, null);
+                                break;
+                            case 10:
+                                channelId = CHANNEL_TYPE_AMBIENTTEMP;
+                                itemType = "Number:Temperature";
+                                addChannel(channelId, itemType, null);
+                                break;
+                            case 37:
+                                channelId = CHANNEL_TYPE_OUTDOORTEMP;
+                                itemType = "Number:Temperature";
+                                addChannel(channelId, itemType, null);
+                                break;
+                        }
+                    }
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            logger.trace("Start Refresh Job");
+            refreshJob = scheduler.scheduleWithFixedDelay(this::getAllUidValues, 0, INTESIS_REFRESH_INTERVAL_SEC,
+                    TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    public void addChannel(String channelId, String itemType, @Nullable final Collection<?> options) {
+        if (thing.getChannel(channelId) == null) {
+            logger.trace("Channel '{}' for UID to be added", channelId);
+            ThingBuilder thingBuilder = editThing();
+            final ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, channelId);
+            Channel channel = ChannelBuilder.create(new ChannelUID(getThing().getUID(), channelId), itemType)
+                    .withType(channelTypeUID).withKind(ChannelKind.STATE).build();
+            thingBuilder.withChannel(channel);
+            updateThing(thingBuilder.build());
+
+            if (getThing().getChannel(channelId) != null) {
```

Comment:
You add this channel to the thing in the code preceding this, so would this ever be false?

Reply:
You're right, removed superfluend null check